ipe_process_orchestrator/assignment_flow.py

```diff
@@ -0,0 +1,134 @@
+        self.rubric_id: str = rubric_id
+        self.course_id: int = course_id
+        self.api_handler: APIHandler = api_handler
+
+    def _look_up_ipe_assignment(self) -> Union[List[int], int]:
+        """
+        This is the use case for copied course that might have the same IPE assignment/group names as the original course.
+        Just look up assignment group by name and then look up the assignment.
+        Will be fetching 100 records only and it is highly unlikely that more than 100 assignment groups will there.
+        the assignment groups will also bring in assignment names to check and compare.
+        """
+        lookup_ag_payload: Dict = {'include[]': 'assignments', 'per_page': 100}
+        lookup_ag_url: str = f'{CANVAS_URL_BEGIN}/courses/{self.course_id}/assignment_groups'
+        lookup_ag_resp: Optional[Response] = self.api_handler.api_call_with_retries(
+            lookup_ag_url, 'GET', lookup_ag_payload)
+
+        err_msg: str = f"Error looking up assignment group '{ASSIGNMENT_GROUP_NAME}' with assignment name '{ASSIGNMENT_NAME}' for course failed {self.course_id}"
+        response_none_check(lookup_ag_resp, err_msg) 
+
+        lookup_ag_resp_json: List[Dict[str, Any]] = json.loads(lookup_ag_resp.text) #type: ignore
+        ag_list: List[int] = list()
+        for ag in lookup_ag_resp_json:
+            if ag['name'].strip() == ASSIGNMENT_GROUP_NAME:
+                for assignment in ag['assignments']:
+                    if assignment['name'].strip() == ASSIGNMENT_NAME:
+                        self._delete_assignment(assignment['id'])
+                ag_list.append(ag['id'])
+        return ag_list if not ag_list else ag_list[0]
+
+    def _delete_assignment(self, assignment_id: int) -> None:
+        """
+        Deleting the copied course assignment
+        """
+        delete_assignment_url: str = f'{CANVAS_URL_BEGIN}/courses/{self.course_id}/assignments/{assignment_id}'
+        delete_assignment_resp: Optional[Response] = self.api_handler.api_call_with_retries(
+            delete_assignment_url, 'DELETE')
+
+        err_msg: str = f'Error deleting assignment {assignment_id} for course failed {self.course_id}'
+        response_none_check(delete_assignment_resp, err_msg)
+
+        logging.info(
+            f'Deleted assignment: {assignment_id} for course: {self.course_id}')
+
+    def _create_assignment_group(self) -> int:
+        """
+        Creating an assignment group with name 'IPE Competencies' always and locate at the end of the assignment list
+        """
+        ag_payload: Dict = {'name': ASSIGNMENT_GROUP_NAME, 'position': 2000}
+        assignment_group_creation_url: str = f'{CANVAS_URL_BEGIN}/courses/{self.course_id}/assignment_groups'
+        ag_resp: Optional[Response] = self.api_handler.api_call_with_retries(
+            assignment_group_creation_url, 'POST', ag_payload)
+
+        err_msg: str = f'Error creating assignment group for course failed {self.course_id}'
```

Comment:
maybe include the assignment group name inside the error message, like:

Error creating assignment group {ASSIGNMENT_GROUP_NAME} for course failed {self.course_id}

Reply:
Done